flutter-for-android.md

```diff
@@ -0,0 +1,269 @@
+
+For example:
+
+[code] 
+
+**In Android, I can Animate a view by View.animate(), how can I do that to a Widget?**
+In Flutter animating widgets can be done easily via a handful of Animation Widgets. See https://flutter.io/widgets/animation/
+
+Lets take a look at how to write a FadeTransition
+
+[code]
+
+**How do I build custom Widgets?**
+
+Building custom widgets in Flutter is very similar to building custom Views in Android. You can extend the Widget and add your custom logic. 
+
+Lets take a look at how to build a custom button
+
+[code]
+
+# **Intents**
+
+**What is the alternative of an Intent in Flutter?**
+
+Flutter does not have the concept of Intents. However in Android there are two main use-cases for Intents, One is to switch between Activities and another is to invoke external(or internal) components of your app (Camera, Services etc).
+
+To switch between screens in Flutter you can directly access the router to draw a new Widget.
+
+[code]
+
+Now in Android you can pass some data along with the Intent, while the Navigation router does not have a way to pass data directly, you can approach this problem by creating a static “data” class that you can store and retrieve values when switching screens. 
+
+There is also a very popular Router called “Fluro” that adds this functionality for you.
+
+[code]
+
+The other popular use-case for Intents is to call external components such as a Camera or File picker. For this you would need to create a native platform integration (or use an existing library)
+
+See [Flutter Plugins] to learn how to build a native platform integration.
+
+
+
+**How do I handle incoming Intents from external applications in Flutter?** 
+
+This can be achieved by registering the intent you want to handle in the AndroidMainfest.xml to launch MainActivity (the default activity created by Flutter)
+
+[code]
+
+Then in MainActivity you can handle the intent and pass it to Flutter 
+
+[code]
+
+Lastly in Flutter you can receive this via
+
+[code]
+
+# **Async UI**
+
+**What is the equivalent of runOnUiThread in Flutter?**
+
+Because of Flutters reactive architecture, there is no need to worry about if you are running stuff on the UI thread, as long as you make sure you dont block your network calls.
+
+For example when making long running calls you should always follow the async/await pattern
+
+[code]
+
+To update the UI you would call setState which would trigger the build method to run again and update the data.
+
+[code]
+
+
+
+**What is the equivalent of an AsyncTask or IntentService on Android?**
+
+In many cases you do not have to worry about threading in Flutter as long as you follow the async/await patterns. 
+
+Flutter built on Dart uses the same execution model of a single threaded event-loop, similar Node.js. To run code asynchronously you can declare the function as an async function and await on long running tasks in the function
+
+[code]
+
+This is where you would typically do network or database calls.
```

Comment:
I think you should be clear: an async task is NOT necessarily run on another thread. It's just a task/function that will run "later" on the main thread.

Some libraries, such as Network and I/O will use another thread (under the covers) until an I/O event is ready, and then the main event loop will be notified.

We don't want people to think if they put a function into a Future, it'll be run on another thread.